tests/testthat/test-renderer2-colour.R

```diff
@@ -0,0 +1,275 @@
+#
+viz.point <- list(pointone = ggplot() + geom_point(
+  data = mtcars,
+  size = 10,
+  aes(x=wt, y=mpg,
+      colour = disp),
+  clickSelects = "gear")+
+  ggtitle("default alpha_off(0.5) style"),
+
+pointtwo = ggplot() + geom_point(
+  data = mtcars,
+  colour="red",
+  colour_off="transparent",
+  size = 10,
+  aes(x=wt, y=mpg,
+      fill = disp),
+  clickSelects = "gear")+
+  ggtitle("colour=\"red\", colour_off=\"transparent\" "),
+
+pointthree = ggplot() + geom_point(
+  data = mtcars,
+  alpha_off=0.5,
+  colour="red",
+  colour_off="transparent",
+  size = 10,
+  aes(x=wt, y=mpg,
+      fill = disp,
+      id=paste0("pointthree_disp", disp, "gear", gear, "wt", wt)),
+  clickSelects = "gear")+
+  ggtitle("colour_off + alpha_off"))
+
+info2 <- animint2HTML(viz.point)
+
+test_that("color_off only changes colour/stroke when clicked, fill does not change", {
+  point.xpath <- '//svg[@id="plot_pointthree"]//circle[@id="pointthree_disp275.8gear3wt3.73"]'
+  circle.list <- getNodeSet(info2$html, point.xpath)
+  before.click.color <- getStyleValue(info2$html, point.xpath, "stroke")
+  before.click.fill <- getStyleValue(info2$html, point.xpath, "fill")
+  
+  clickID('pointthree_disp275.8gear3wt3.73')
+  html <- getHTML()
+  after.click.color <- getStyleValue(html, point.xpath, "stroke")
+  after.click.fill <- getStyleValue(html, point.xpath, "fill")
+  
+  expect_false(isTRUE(all.equal(before.click.color, after.click.color)))
+  expect_equal(before.click.fill, after.click.fill)
+})
+
+test_that("fill and color are not same", {
+  point.xpath <- '//svg[@id="plot_pointthree"]//circle[@class="geom"]'
+  circle.list <- getNodeSet(info2$html, point.xpath)
+  circle.color <- getStyleValue(info2$html, point.xpath, "stroke")
+  circle.fill <- getStyleValue(info2$html, point.xpath, "fill")
+  expect_false(isTRUE(all.equal(circle.color, circle.fill)))
+})
+
+#
+# tests for g$geom="rect", originally only support 'stroke' as selection style
+#
+row.vec <- paste("row", c(1:3))
+col.vec <- paste("col", c(1:3))
+heat.data <- data.table(row.name = row.vec,
+  col.name = rep(col.vec, each=length(row.vec)),
+  value = c(2,8,-5,-7,15,3,-1,6,-7.5))
+no.col <- ggplot() +
+  geom_tile(data=heat.data,
+  aes(x = row.name, y = col.name, fill = value,
+  id=paste0("no_col_", value)),
+  clickSelects = "value")
+has.col.no.off <- ggplot() +
+  geom_tile(data=heat.data,
+  aes(x = row.name, y = col.name, fill = value,
+  id=paste0("col_", value)),
+  colour="red",
+  clickSelects = "value")
+has.col.and.off <- ggplot() +
+  geom_tile(data=heat.data,
+  aes(x = row.name, y = col.name, fill = value,
+  id=paste0("col_off_", value)),
+  colour="red",
+  colour_off="grey50",
+  clickSelects = "value")
+viz.tile <- list(
+  nocol=no.col,
+  colnooff=has.col.no.off,
+  colandoff=has.col.and.off)
+
+info <- animint2HTML(viz.tile)
+
+test_that("if has clickSelects but no colour/colour_off, selection colour/stroke should be black, and transparent for not selected (no stroke)", {
```

Comment:
hey @tdhock , I added some tests here including these situations on `geom_tile`:
1. clickSelects ✔️ , colour ✖️  , colour_off ✖️ : not selected = **transparent**, selected = **black**;
2. clickSelects ✔️ , colour ✔️  , colour_off ✖️ : not selected = **transparent**, selected = **colour** value;
3. clickSelects ✔️ , colour ✔️  , colour_off ✔️ : not selected = **colour_off**, selected = **colour**;

See the the [visualization here](https://bl.ocks.org/Faye-yufan/raw/4a2e093011b9213b223b2850b28a239a/), please take a look at them.

Reply:
that is a great test case. maybe for ease of visual inspection, can you please make the size bigger? I guess that is the default size=2, so maybe increase to 5 or so?